### PR TITLE
[13.0][IMP] account_invoice_report_grouped_by_picking: Show section and notes

### PR DESCRIPTION
**Pull request by carlosdauden**
- Compatibility with section and notes lines (display_type)
 - Remove roadmap
 - Fix last_picking position to avoid picking header repeat in first line.

@Tecnativa TT39052

### CODE REVIEW

**Comment by carlosdauden** [comment]
ping @chienandalu @sergio-teruel

File: account_invoice_report_grouped_by_picking/README.rst [hunk 1]
```diff
@@ -14,13 +14,13 @@ Account Invoice Grouped by Picking
     :target: http://www.gnu.org/licenses/agpl-3.0-standalone.html
     :alt: License: AGPL-3
 .. |badge3| image:: https://img.shields.io/badge/github-OCA%2Faccount--invoice--reporting-lightgray.png?logo=github
-    :target: https://github.com/OCA/account-invoice-reporting/tree/13.0/account_invoice_report_grouped_by_picking
+    :target: https://github.com/OCA/account-invoice-reporting/tree/15.0/account_invoice_report_grouped_by_picking
```
Comment (sergio-teruel) [inline account_invoice_report_grouped_by_picking/README.rst]: Is for 13.0 version... ¿?

**Review by chienandalu — APPROVED** [review]
:+1:

**Comment by pedrobaeza** [comment]
On the runboat:

http://oca-account-invoice-reporting-13-0-pr258-75118eca2170.runboat.odoo-community.org/web#id=10&action=318&active_id=2&model=account.move&view_type=form&cids=&menu_id=231

I'm getting the error:

```
AttributeError: 'bool' object has no attribute 'price_subtotal'


Error to render compiling AST
AttributeError: 'bool' object has no attribute 'price_subtotal'
Template: account.report_invoice_document_with_payments
Path: /t/t/div/table/tbody/t[3]/t[7]
Node: <t t-set="current_subtotal" t-value="current_subtotal + line.price_subtotal" groups="account.group_show_line_subtotals_tax_excluded"/>
```

**Comment by carlosdauden** [comment]
Link shows "503 Service Unavailable".
I add commit with changes, can you try now?

**Comment by pedrobaeza** [comment]
I still have the same error:

```
AttributeError: 'bool' object has no attribute 'price_subtotal'

Error to render compiling AST
AttributeError: 'bool' object has no attribute 'price_subtotal'
Template: account.report_invoice_document_with_payments
Path: /t/t/div/table/tbody/t[3]/t[7]
Node: <t t-set="current_subtotal" t-value="current_subtotal + line.price_subtotal" groups="account.group_show_line_subtotals_tax_excluded"/>
```

You can try it on http://oca-account-invoice-reporting-13-0-pr258-2a0f03256261.runboat.odoo-community.org/web#id=9&action=318&active_id=1&model=account.move&view_type=form&cids=&menu_id=231

**Comment by carlosdauden** [comment]
Fixed

**Comment by pedrobaeza** [comment]
Now I get this error:
```
Traceback (most recent call last):
  File "/opt/odoo/odoo/addons/base/models/qweb.py", line 334, in _compiled_fn
    return compiled(self, append, new, options, log)
  File "<template>", line 1, in template_account_report_invoice_document_with_payments_306
  File "<template>", line 2, in body_call_content_304
  File "<template>", line 95, in foreach_271
TypeError: unsupported operand type(s) for +: 'NoneType' and 'int'

Error to render compiling AST
TypeError: unsupported operand type(s) for +: 'NoneType' and 'int'
Template: account.report_invoice_document_with_payments
Path: /t/t/div/table/tbody/t[3]/t[9]
Node: <t t-if="current_section and (line_last or lines[line_index+1].display_type == 'line_section')">
                                    <tr class="is-subtotal text-right">
                                        <td colspan="99">
                                            <strong class="mr16">Subtotal</strong>
                                            <span t-esc="current_subtotal" t-options="{&quot;widget&quot;: &quot;monetary&quot;, &quot;display_currency&quot;: o.currency_id}"/>
                                        </td>
                                    </tr>
                                </t>
```
You can check it in http://oca-account-invoice-reporting-13-0-pr258-2a66aa7ac692.runboat.odoo-community.org/web#id=9&action=318&active_id=1&model=account.move&view_type=form&cids=&menu_id=231 (if still available when checked).

**Comment by pedrobaeza** [comment]
@carlosdauden this still has a problem.

**Comment by carlosdauden** [comment]
Improved compatibility with other modules.
Tested on runbot with this invoice:
http://oca-account-invoice-reporting-13-0-pr258-ed3421d54152.runboat.odoo-community.org/web#id=10&action=318&active_id=2&model=account.move&view_type=form&cids=&menu_id=231

**Comment by AsierAcedo** [comment]
Hi @carlosdauden check this invoice:
[http://oca-account-invoice-reporting-13-0-pr258-ed3421d54152.runboat.odoo-community.org/web#id=11&action=318&model=account.move&view_type=form&cids=&menu_id=162](http://oca-account-invoice-reporting-13-0-pr258-ed3421d54152.runboat.odoo-community.org/web#id=11&action=318&model=account.move&view_type=form&cids=&menu_id=162)

On the report the first note is wrong.

**Review by pedrobaeza — APPROVED** [review]
Tested and working

/ocabot merge minor

**Comment by HaraldPanten** [comment]
@pedrobaeza You didn't port these changes to V14, right? I think they're in V15 and V16 but not in V14 🤔

**Comment by pedrobaeza** [comment]
It may be. Please check in the commit history.

**Pull request merged**